TFT/src/User/API/Gcode/mygcodefs.c

```diff
@@ -140,23 +140,22 @@ bool scanPrintFilesGcodeFs(void)
         continue;
       }*/
 
-      infoFile.file[infoFile.fileCount] = malloc(strlen(file) + 2);  // plus one extra byte for filename extension check
+      infoFile.file[infoFile.fileCount] = malloc(strlen(file) + 2);  // "+ 2": space for terminating null character and the flag for filename extension check
       if (infoFile.file[infoFile.fileCount] == NULL)
       {
         if (infoFile.longFile[infoFile.fileCount] != 0)
           free(infoFile.longFile[infoFile.fileCount]);
 
         break;
       }
-
-      strcpy(infoFile.file[infoFile.fileCount], file);
-      infoFile.file[infoFile.fileCount][strlen(file) + 1] = 0;  // set to 0 the extra byte for filename extension check
+      // copy file name and set the flag for filename extension check
+      strncpy(infoFile.file[infoFile.fileCount], file, strlen(file) + 2);  // "+2":  terminating null character and the flag for filename extension check
```

Comment:
// "+ 2": space for terminating....     instead of  // "+2":  terminating....

Reply:
Done